changelogs/fragments/7698-improvements-to-keycloak_realm_key.yml

```diff
@@ -0,0 +1,5 @@
+minor_changes:
+  - keycloak_realm_key module now supports rsa enc key usage (`rsa-enc`) via `provider_id` parameter (https://github.com/ansible-collections/community.general/pull/7698).
+  - keycloak_realm_key module now supports 8 additional key algorithms via `config.algorithm` parameter (https://github.com/ansible-collections/community.general/pull/7698).
+  - keycloak_realm_key module now sets a default empty string for `config.certificate` parameter instead of forcing user to specify a value (https://github.com/ansible-collections/community.general/pull/7698).
```

Comment:
What's the rationale behind this change? In the PR description you mention:

> fixes a bug where the user must specify a certificate (no longer required)

but I don't see how this is a bug?

Reply:
Hi @felixfontein, the reason that I considered it a bug is because the very first example listed suggests that it is not required (so perhaps an error in docs):
```
EXAMPLES = '''
- name: Manage Keycloak realm key (certificate autogenerated by Keycloak)
  community.general.keycloak_realm_key:
    name: custom
    state: present
    parent_id: master
    provider_id: rsa
    auth_keycloak_url: http://localhost:8080/auth
    auth_username: keycloak
    auth_password: keycloak
    auth_realm: master
    config:
      private_key: "{{ private_key }}"
      enabled: true
      active: true
      priority: 120
      algorithm: RS256
```

Reply:
The problem is that adding a default makes it pretty hard to change the behavior of not specifying the value later - for example so that not specifying a value doesn't change the certificate if the key doesn't change.

It's better to fix the docs instead (best done in a separate PR so it can be backported further - features can only end up in the latest `stable-x` branch, but docs fixes can go back to all maintained `stable-x` branches.).